openssl/src/hash.rs

```diff
@@ -796,14 +796,17 @@ mod tests {
         }
 
         assert_eq!(MessageDigest::shake_128().block_size(), 168);
+        #[cfg(ossl111)]
         assert_eq!(MessageDigest::shake_128().size(), 16);
+        #[cfg(awslc)]
+        assert_eq!(MessageDigest::shake_128().size(), 0);
```

Comment:
This seems wrong?

Reply:
The SHAKE algorithms are extendable output algorithms that don't have an explicit digest output size as it's determined based on the requested output size. So we opt to return `0` in this instance if `EVP_MD_size()` is called on such an XOF algorithm like SHAKE-128 and SHAKE-256.